### PR TITLE
[WEB-3840] remove Cypress in favour of Storybook's test-runner

### PR DESCRIPTION
**Pull request by jamiehenson**
## Jira Ticket Link / Motivation

https://ably.atlassian.net/browse/WEB-3840

## Summary of changes

Since the Storybook migration (pre v14), Cypress has been inactive. Upon reflection, now we have Storybook, Cypress seems a somewhat heavy handed approach to testing when we don't need deep integration testing - something like snapshot testing would probably be a good starting point but there wasn't a clear resolution to the conversation.

Thankfully for us, Storybook has options for all of the above. With its own `test-runner`, we get smoke tests (i.e. does this component render or not) for free for each story, and we have the option of writing more comprehensive testing journeys using [Play functions](https://storybook.js.org/docs/writing-stories/play-function) - which given that they're written alongside the stories themselves seems more ergonomic.

This PR brings in `test-runner`, guts all the `cypress` stuff, and makes some adjustments to the Github workflows so that the test suite runs on push.

## How do you manually test this?

Pull down, run `yarn test`, Storybook should build and the test suite should run, resulting in a successful set of smoke tests like this

Screenshot 2024-06-11 at 16 25 43

## Merge/Deploy Checklist

- [ ] Written automated tests for implemented features/fixed bugs
- [ ] Rebased and squashed commits
- [ ] Commits have clear descriptions of their changes
- [ ] Checked for any performance regressions

## Frontend Checklist

- [ ] No frontend changes in this PR
- [ ] Added before/after screenshots for changes
- [ ] Tested on different platforms/browsers with [Browserstack](https://www.browserstack.com)
- [ ] Compared with the initial design / our [brand guidelines](https://www.figma.com/file/jTQgyIovrhGBOf4Zrvjvbk/Ably-Linear-Design?node-id=118%3A0)
- [ ] Checked the code for accessibility issues ([VoiceOver User Guide](https://support.apple.com/en-gb/guide/voiceover/welcome/mac))?

### CODE REVIEW

File: .swc [hunk 1]
```diff
@@ -1,13 +1,13 @@
 {
   "jsc": {
-    "target": "es2018",
+    "target": "es2021",
```
Comment (jamiehenson) [inline .swc:3]: This aligns with Node 16, which is the minimum requirement across our other apps

**Review by kennethkalmer — APPROVED** [review]
This looks great @jamiehenson, thanks!

Just an aside, the tests for the `Status` component is definitely flaky and we will get false negatives. This is because it goes from a _loading_ state to whatever the story dictates, and depending on the time of the snapshot it might still be doing its _animated pulse_ rather than rendering the status colour. I don't think it should block us for now, just as long as we're aware :)

**Review by aralovelace — APPROVED** [review]
cant wait for this

**Comment by jamiehenson** [comment]
@kennethkalmer I removed the "Live" story on `Status` which was causing some flakiness, but the others seemed to be well stubbed so not expecting further flakiness - might not have gone far enough though

**Pull request merged**